conceptual.md

```diff
@@ -3,28 +3,46 @@
 Answer the following questions below:
 
 - What is a JWT?
+  * JSON Web Token - open standard that allows tokens to be shared between parties like a server and a client
 
 - What is the signature portion of the JWT?  What does it do?
+  * The signature is the third portion of the token string. Allows the receiving party to verify the sender is who they say they are. Also ensures the data of the payload or headers haven't been tampered with as they are used in calculating the signature.
 
 - If a JWT is intercepted, can the attacker see what's inside the payload?
+  * Yes
 
 - How can you implement authentication with a JWT?  Describe how it works at a high level.
+  * After you log a user in on your server, you can generate and respond with a JWT that includes info like the username or a property like "isAuthenticated". You would then include this token in any requests back to the server from the client. The server can then respond if the token is verified or deny the request if it fails verification.
 
 - Compare and contrast unit, integration and end-to-end tests.
+  * Unit tests are small, isolated tests that test only a specific function or unit of code.
+  * Integration tests are for testing that two or more units are able to work together. An example would be testing an API endpoint as this tests not only that the API responds properly, but that the API code is able to retrieve data correctly like getting it from a database.
+  * End-to-end tests are for testing an application from a user perspective. This could include things like clicking a login button, filling out the login form, clicking the button to submit, and being shown either the logged in screen if successful or some field validation if a field was filled out incorrectly.
 
 - What is a mock? What are some things you would mock?
+  * Mocks are replacing dependency objects like functions or data in a test when it's not practical to have it included. You might mock the result of a function or data that would be returned from an external API.
```

Comment:
They're primarily used to unit test a function as opposed to having to integrate with an external API.